blog/2023-12-12-magazine-007.md

```diff
@@ -0,0 +1,291 @@
+
+A very cool [winglib](https://github.com/winglang/winglibs) was born and already [making the
+rounds](https://www.linkedin.com/posts/marcio-almeida-25891669_winglang-cloud-activity-7146152178737463296-s5gW)
+on the internets.
+
+You can now use the `websockets` library to deploy a WebSocket server. This works in the Wing
+Simulator and supported on the AWS platforms (`tf-aws` and `awscdk`) backed by API Gateway V2.
+
+Check out the library's
+[README](https://github.com/winglang/winglibs/blob/main/websockets/README.md) for more details and
+examples.
+
+```sh
+npm i @winglibs/websockets
+```
+
+Here's a stupid little example:
+
+```js
+bring websockets as ws;
+
+let server = new ws.WebSocket(name: "my-server");
+
+server.onConnect(inflight (id) => {
+  log("client {id} connected");
+});
+
+server.onDisconnect(inflight (id) => {
+  log("client {id} disconnected");
+});
+
+server.onMessage(inflight (id, message) => {
+  log("client {id} sent {message}");
+});
+```
+
+Common use case is tracking and maintaining connection pool.
+
+```js
+bring ex;
+
+let conections = new ex.DynamodbTable(
+  name: "Connections",
+  hashkey: "id",
+  attributeDefinitions: {
+    "id": "S",
+  },
+};
+```
+
+And here are few usages:
+
+```js
+ws.onConnect(inflight(id) => {
+  connections.putItem({
+    item: {"id": id }
+  });
+});
+
+ws.onDisconnect(inflight(id) => {
+  connections.deleteItem({
+    key: {"id": id }
+  });
+});
+
+ws.onMessage(inflight(id, body) => {
+  for conn in tb.scan().Items {
+    ws.sendMessage(str.fromJson(conn.get("id")), body);
+  }
+});
+```
+
+❤️ [Marcio Cruz](https://github.com/marciocadev)
+
+### Upcoming Events
+
+**🔥 Workshop: Wing + Amazon Bedrock = :heart: - January 3rd @4:30 - 6pm IST**
+
+Get your hands dirty with [Eyal Keren](https://github.com/ekeren) as we build together an
+application that allows you to load any PDF file and ask questions about it using one of [Amazon
+Bedrock](https://aws.amazon.com/bedrock/)'s generative AI models.
+
+[Reserve your spot](https://www.eventbrite.com/e/amazon-bedrock-winglang-tickets-769562721817?aff=oddtdtcreator).
+
+### The Wingly Update
+
+**[The Wingly Update](https://www.twitch.tv/winglangio)** is our weird biweekly stream where we
+share the latest developments of the project, chat with folks from the cloud industry, geek out and
+celebrate the beauty of the cloud.
+
+If you haven't been able to catch our show, you can find the complete stack of [all our episodes
+here](https://youtube.com/playlist?list=PL-P8v-FRassZBWsNoSafL_ReO0JO0xJVm&si=trffVrtGGMUZ-SKb). 
+
+Here are a few segments from the previous shows we thought might be interesting to check out:
+
+- [The changelog segment](TBD): In this clip from The Wingly Update #25, [Mark](TBD) walks us through the recent changes! Thank you 16 Wingnuts for 37 wingdiffs!
+- [Creating a FIFO Queue](TBD): [Eyal](https://github.com/ekeren) and [Elad](https://github.com/eladcohen) are walking us through on how to create a *FIfo* Queue!
```

Comment:
I think there must be a few more interesting segments to share. The changelog is not a good one to include it generally.

Reply:
We can combine the latest Wingly and have more segments... Is that what you have in mind?
We should also add Eyal's workshop - but in a different location here

Reply:
I think there are many segments from older Wingly episodes that are already in our YouTube that we can put here instead of the changelog.